purchases/src/main/kotlin/com/revenuecat/purchases/deeplinks/DeepLinkHandler.kt

```diff
@@ -0,0 +1,71 @@
+package com.revenuecat.purchases.deeplinks
+
+import android.content.Intent
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.verboseLog
+
+/**
+ * Allows to pass deep links to the RevenueCat SDK for processing.
+ */
+@ExperimentalPreviewRevenueCatPurchasesAPI
+class DeepLinkHandler {
+    /**
+     * Represents whether the deep link will be handled by the SDK or not.
+     */
+    enum class HandleResult {
+        /**
+         * Deep link is recognized by RevenueCat's SDK and will be handled.
+         */
+        HANDLED,
+
+        /**
+         * Deep link is recognized by RevenueCat's SDK but it's not configured so it will be cached
+         * and processed after SDK has been configured.
+         */
+        DEFERRED_TO_SDK_CONFIGURATION,
+
+        /**
+         * Deep link is not recognized by RevenueCat's SDK so it's ignored.
+         */
+        IGNORED,
+    }
+
+    companion object {
+        @get:Synchronized
+        internal val cachedLinks = mutableSetOf<DeepLinkParser.DeepLink>()
```

Comment:
Changed to use a "synchronized set" instead... Basically want to avoid someone clicking on the link twice and adding it twice to the queue... As for handling in order, I don't believe it's a big deal even if processed out of order at least for now. Lmk if you think otherwise!

Reply:
I think that works! I'm not sure if handling out of order is a problem. 🤔 If we think it might be, we could look into using a `LinkedHashSet`. Otherwise I think this is fine.